apps/astarte_appengine_api/test/astarte_appengine_api_web/controllers/interface_controller_test.exs

```diff
@@ -217,6 +217,64 @@ defmodule Astarte.AppEngine.APIWeb.InterfaceControllerTest do
                )
       end
     end
+
+    property "deletes data when is possible", context do
+      %{realm_name: realm_name, interfaces: interfaces, device: device, auth_conn: conn} = context
+
+      valid_interfaces_to_unset =
+        interfaces
+        |> Enum.filter(&(&1.ownership == :server && &1.type == :properties))
+
+      check all interface <- member_of(valid_interfaces_to_unset),
+                mapping_to_unset <- member_of(interface.mappings),
+                path <- path_from_endpoint(mapping_to_unset.endpoint) do
+        if mapping_to_unset.allow_unset do
+          path_tokens = path_tokens(path)
+          expected_token = [realm_name, device.encoded_id, interface.name | path_tokens]
+
+          publish_result_ok(interface, mapping_to_unset, fn args ->
+            assert %{payload: payload, topic_tokens: topic_tokens, qos: qos} = args
+            assert topic_tokens == expected_token
+            assert qos == 2
+            assert payload == ""
+          end)
+        end
+
+        request_path =
+          interface_values_path(
+            conn,
+            :delete,
+            realm_name,
+            device.encoded_id,
+            interface.name,
+            path_tokens(path)
+          )
+
+        conn = delete(conn, request_path)
+
+        case mapping_to_unset.allow_unset do
+          false ->
+            assert response(conn, 400)
+
+          true ->
+            assert response(conn, 204)
```

Comment:
Nit: would it be possible to write two separate tests, one for `allow_unset: true` and one for `allow_unset: false`? This way, we clearly separate the two different flows

Reply:
It would be pretty difficult as we're using the device case to install a device and its interfaces and then we select one among those to perform an update, we do not generate the interfaces and devices directly here

Reply:
Done :+1: